### PR TITLE
refactor: move from `io/ioutil` to `io` and `os` packages

### PR DESCRIPTION
**Pull request by Juneezee**
### What does this PR do?

The `io/ioutil` package has been deprecated in Go 1.16 (See https://pkg.go.dev/io/ioutil). This PR replaces the existing `io/ioutil` functions with their new definitions in `io` and `os` packages.

- `ioutil.Discard` => `io.Discard`
- `ioutil.NopCloser` => `io.NopCloser`
- `ioutil.ReadAll` => `io.ReadAll`
- `ioutil.ReadDir` => `os.ReadDir` (returns a slice of `os.DirEntry` rather than a slice of `fs.FileInfo`, this may improve performance in some cases as `os.ReadDir` is more efficient [\[1\]])
- `ioutil.ReadFile` => `os.ReadFile`
- `ioutil.TempDir` => `os.MkdirTemp`
- `ioutil.TempFile` => `os.CreateTemp`
- `ioutil.WriteFile` => `os.WriteFile`

[\[1\]]: https://pkg.go.dev/io/ioutil#ReadDir

### Motivation

Code cleanup to replace deprecated functions.

### Additional Notes

No user-facing changes introduced.

### Describe how to test/QA your changes

All tests should continue to pass. `go build` should not fail.

### Reviewer's Checklist

- [ ] If known, an appropriate milestone has been selected; otherwise the `Triage` milestone is set.
- [ ] Use the `major_change` label if your change either has a major impact on the code base, is impacting multiple teams or is changing important well-established internals of the Agent. This label will be use during QA to make sure each team pay extra attention to the changed behavior. For any customer facing change use a releasenote.
- [ ] A [release note](https://github.com/DataDog/datadog-agent/blob/main/docs/dev/contributing.md#reno) has been added or the `changelog/no-changelog` label has been applied.
- [ ] Changed code has automated tests for its functionality.
- [ ] Adequate QA/testing plan information is provided if the `qa/skip-qa` label is not applied.
- [ ] At least one `team/..` label has been applied, indicating the team(s) that should QA this change.
- [ ] If applicable, docs team has been notified or [an issue has been opened on the documentation repo](https://github.com/DataDog/documentation/issues/new).
- [ ] If applicable, the `need-change/operator` and `need-change/helm` labels have been applied.
- [ ] If applicable, the `k8s/<min-version>` label, indicating the lowest Kubernetes version compatible with this feature. 
- [ ] If applicable, the [config template](https://github.com/DataDog/datadog-agent/blob/main/pkg/config/config_template.yaml) has been updated.

### CODE REVIEW

**Review by mellon85 — APPROVED** [review]
Remote Config LGTM

**Review by clarkb7 — APPROVED** [review]

**Review by juliogreff — APPROVED** [review]
Looks good for container-integrations

**Review by brycekahle — APPROVED** [review]
👍 for ebpf-platform files

**Review by hmahmood — APPROVED** [review]

**Review by dineshg13 — APPROVED** [review]

**Review by cedricvanrompay-datadog — APPROVED** [review]

**Review by sgnn7 — APPROVED** [review]

**Comment by Juneezee** [comment]
Rebased and resolved conflicts in
```
pkg/network/tracer/tracer_test.go
pkg/secrets/check_rights_windows_test.go
```

**Review by guyarb — APPROVED** [review]
/LGTM for Universal Service Monitoring files

File: pkg/forwarder/internal/retry/on_disk_retry_queue.go [hunk 1]
```diff
@@ -229,16 +228,21 @@ func (s *onDiskRetryQueue) reloadExistingRetryFiles() error {
...
 	var files []os.FileInfo
 	currentSizeInBytes := int64(0)
 	for _, entry := range entries {
-		if entry.Mode().IsRegular() && filepath.Ext(entry.Name()) == retryTransactionsExtension {
-			currentSizeInBytes += entry.Size()
-			files = append(files, entry)
+		info, err := entry.Info()
+		if err != nil {
+			continue
+		}
```
Comment (remeh) [inline pkg/forwarder/internal/retry/on_disk_retry_queue.go:241]: Let's not silent this one.

```suggestion
		if err != nil {
			log.Warn("Can't get file info", err)
			continue
		}
```

**Review by remeh — APPROVED** [review]
Thanks a lot for your contribution! Left a small suggestion but LGTM from @DataDog/agent-metrics-logs perspective.

**Review by mx-psi — APPROVED** [review]

**Review by sarah-witt — APPROVED** [review]
Looks good for platform integrations!

**Review by mbotarro — APPROVED** [review]
Looks good for @DataDog/processes

**Review by AlexandreYang — APPROVED** [review]

**Review by ahmed-mez — APPROVED** [review]

**Review by leeavital — APPROVED** [review]

**Review by maxday — APPROVED** [review]
LGTM for `serverless` related changes

**Review by fisherevans — APPROVED** [review]
looks good for container-apps - thanks :D

**Pull request merged**